src/details/ArborX_Point.hpp

```diff
@@ -34,7 +34,7 @@ class Point
   }
 
   KOKKOS_INLINE_FUNCTION
-  constexpr Point(Data data)
+  constexpr Point(Data const &data)
```

Comment:
I would have passed by value and moved into place.

Reply:
Anyway I see little value in the PR overall besides the couple places we had forgotten double braces to silent warnings

Reply:
I see value in making sure that the underlying values are initialized, but won't fight for this pull request if no one else supports it.

Reply:
Then fix your constructor.  Your current implementation makes us pay for an extra copy if we pass a rvalue.

Reply:
The assembly is the same (https://godbolt.org/z/ePlQOS) but I also like the value variant better.